lego/apps/events/serializers/events.py

```diff
@@ -247,8 +247,12 @@ def get_photo_consents(self, obj: Event):
         # Only return consents for events that use consent
         # and the user is allowed to register
         if (
-            not obj.use_consent
-            or len(obj.get_possible_pools(request.user, future=True)) == 0
+            not obj.is_admitted(request.user)
+            and not obj.is_on_waiting_list(request.user)
+            and (
+                not obj.use_consent
+                or len(obj.get_possible_pools(request.user, future=True)) == 0
+            )
```

Comment:
Just when I though this couldn't get any uglier ... It's impossible to read (in combination with the check above) and I'll just have to trust that it works ... Not to mention that further developing this is a pain in the ass

Reply:
Hmm, yeah maybe you're right :grimacing:. What about this:
```python
if (
    not obj.use_consent or (
        not obj.is_admitted(request.user)
        and not obj.is_on_waiting_list(request.user)
        and len(obj.get_possible_pools(request.user, future=True)) == 0
    )
 )
```

I also wanted to extract the `not obj.is_admitted` into a separate if, but the peeps at the office preferred this.

Also, there is a bug here as you can see :upside_down_face:

Reply:
@ivarnakken I changed it to the suggestion here now. Let me know if you still think it should be different, otherwise I'll merge this ASAP.

Reply:
Sure, this looks better. I think `not obj.is_admitted(request.user)` should be separated though

Reply:
Done